### ISSUE TITLE
[Bug]: TestGossipInSimulatedCluster goroutine leak

### ISSUE THREAD

**Issue by zhangxu19830126** [matrixorigin/matrixone#4992]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

Goroutine leak in memberlist lib

### Expected Behavior

No goroutine leak

**Comment by zhangxu19830126** [comment]
@volgariver6 TestGossipInSimulatedCluster has a goroutine leak.  Now the test is skipped,  reopen after you fixed the leak.

**Comment by volgariver6** [comment]
Have tried more than 500 times in local env but cannot repro the leak issue. So submit a PR to remove the test skip to repro it in CI and get the leak stack.

**Comment by volgariver6** [comment]
Found the leak stack:

```
2022-11-24T06:00:27.9623530Z     leak.go:138: Leaked goroutine: goroutine 19570 [chan receive]:
2022-11-24T06:00:27.9623890Z         github.com/hashicorp/memberlist.(*Memberlist).probeNode(0xc0029a4b00, 0xc01c493200)
2022-11-24T06:00:27.9624580Z            /Users/runner/go/pkg/mod/github.com/matrixorigin/memberlist@v0.4.1-0.20221101065119-bb7bce164406/state.go:470 +0x1cce
2022-11-24T06:00:27.9624890Z         github.com/hashicorp/memberlist.(*Memberlist).probe(0xc0029a4b00)
2022-11-24T06:00:27.9625570Z            /Users/runner/go/pkg/mod/github.com/matrixorigin/memberlist@v0.4.1-0.20221101065119-bb7bce164406/state.go:255 +0x388
2022-11-24T06:00:27.9626040Z         github.com/hashicorp/memberlist.(*Memberlist).triggerFunc(0x1000750f9?, 0x14dc9380, 0xc006334f00, 0xc00485a900, 0xc0005ef400)
2022-11-24T06:00:27.9626710Z            /Users/runner/go/pkg/mod/github.com/matrixorigin/memberlist@v0.4.1-0.20221101065119-bb7bce164406/state.go:157 +0x147
2022-11-24T06:00:27.9627010Z         created by github.com/hashicorp/memberlist.(*Memberlist).schedule
2022-11-24T06:00:27.9627690Z            /Users/runner/go/pkg/mod/github.com/matrixorigin/memberlist@v0.4.1-0.20221101065119-bb7bce164406/state.go:121 +0x406
2022-11-24T06:00:27.9627950Z --- FAIL: TestGossipInSimulatedCluster (405.68s)
```

**Comment by volgariver6** [comment]
There are performance issues in this test case, need to do more research to fix this.

**Comment by volgariver6** [comment]
fixed by #7021. This issue could be closed.
